### DISCUSSION TITLE
Is there a hook or DBUS event for unlocking the database?

### DISCUSSION THREAD

**Discussion by BettaGeorge** [keepassxreboot/keepassxc#7770] [Q&A] [answered]
Hello all!

Short version: I would like the ability to execute a bash script after I unlock my database. Is this possible?

Long version: I use keepassxc as my keyring together with secret-tool. When I log into any of my machines, the first thing that starts is the keepassxc "unlock database" prompt, after which everything is great and neatly integrated.

Except my Nextcloud. My current workflow is: log in to my machine, enter the keepassxc master passphrase, hit enter, then manually open up a command prompt and start the Nextcloud client. Nextcloud sees a secret service running, fetches its login information from it, and is happy.

Obviously, that last step is terribly bothersome. However, if I start Nextcloud before the database is unlocked, it finds no secret service running and instantly launches my browser, prompting for a password. Hence I cannot have Nextcloud in my autostart. And because the keepassxc process does (of course) not terminate after unlocking the database, a simple "keepassxc; nextcloud" is no solution either.

Hence I would ideally like to be able to detect when the keepassxc database has been unlocked and react by automatically invoking Nextcloud.

Alternatively, I would settle for teaching Nextcloud to invoke keepassxc instead of opening a browser window.

There is no other keyring or secret service running on my system; I am on Manjaro GNU/Linux with lightdm as my display manager and EXWM as my window manager.

Any advice is appreciated!

**Comment by droidmonkey** [comment, marked as answer]
https://github.com/keepassxreboot/keepassxc/blob/develop/src/gui/org.keepassxc.KeePassXC.MainWindow.xml

**Reply by BettaGeorge** [reply]
That was easy. Thanks!

**Reply by BettaGeorge** [reply]
Actually, this only seems to export methods and no signals. Unless I am missing something? I need a signal that I can intercept to do stuff, not a method I can call to (un)lock the database via DBUS.

**Reply by droidmonkey** [reply]
2.7.0 secret service asks for database unlock if it is not already unlocked

**Reply by BettaGeorge** [reply]
I am not sure that the Nextcloud client knows how to do this -- at least I have not found a setting to do so.

If there is indeed no DBUS signal, I guess I will have to ask on the Nextcloud help forum.
